tinygrad/tensor.py

```diff
@@ -1961,15 +1962,21 @@ def interpolate(self, size:Tuple[int, ...], mode:str="linear", align_corners:boo
     ```
     """
     assert isinstance(size, (tuple,list)) and all_int(size) and 0 < len(size) <= self.ndim, f"invalid {size=}"
-    assert mode == "linear", "only supports linear interpolate"
-    x, expand = self, list(self.shape)
+    assert not (mode == "nearest" and align_corners), 'align_corners option can only be set with the mode="linear"'
+    x, expand, indexes = self, list(self.shape), []
     for i in range(-len(size), 0):
       scale = (self.shape[i] - int(align_corners)) / (size[i] - int(align_corners))
       arr, reshape = Tensor.arange(size[i], dtype=dtypes.float32, device=self.device), [1] * self.ndim
-      index = (scale*arr if align_corners else (scale*(arr+0.5))-0.5).clip(0, self.shape[i]-1)
       reshape[i] = expand[i] = size[i]
-      low, high, perc = [y.reshape(reshape).expand(expand) for y in (index.floor(), index.ceil(), index - index.floor())]
-      x = x.gather(i, low).lerp(x.gather(i, high), perc)
+      if mode in {"nearest", "nearest-exact"}:
+        index = ((arr + (0.5 if mode == "nearest-exact" else 0)) * scale).floor().cast(dtypes.int)
+        indexes.append(index.reshape(reshape[-len(size):]).expand(size))
+      elif mode == "linear":
+        index = (scale*arr if align_corners else (scale*(arr+0.5))-0.5).clip(0, self.shape[i]-1)
+        low, high, perc = [y.reshape(reshape).expand(expand) for y in (index.floor(), index.ceil(), index - index.floor())]
+        x = x.gather(i, low).lerp(x.gather(i, high), perc)
+      else: raise ValueError(f"invalid {mode=}")
+    if mode in {"nearest", "nearest-exact"}: x = x[tuple(slice(None) for _ in range(x.ndim - len(indexes))) + tuple(indexes)]
```

Comment:
also, this line of code looks weird because pylint won't allow `x = x[..., *indexes]`

Reply:
Hmm, why won't pylint allow that?